src/pages/surveys/[id].page.tsx

```diff
@@ -1,10 +1,32 @@
+import type { NextPage } from 'next';
 import { useRouter } from 'next/router';
+import { useContext, useEffect } from 'react';
 
-const SurveyDetail = () => {
+import { getSurveyDetail } from 'services/survey';
+import { ACTIONS, StoreContext } from 'store';
+
+const SurveyDetail: NextPage = () => {
   const router = useRouter();
-  const { id } = router.query;
+  const surveyId = router.query.id as string | undefined;
```

Comment:
I think we can use `getServerSideProps` to get the `surveryId` instead.
You can check it [here](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#when-does-getserversideprops-run).

We can fetch the data from the server side, and pass it directly to the prop of this component.

The advantage of this way is the content can be served faster than fetching from the client-side.

What do you think? 🤔

Reply:
@hoangmirs I looked at this before, but with the current approach I use to make an HTTP request to the backend, it doesn't support `getServerSideProps`. Because the HTTP request requires a user token which stored in local storage, it can't be accessed on the server-side. 
I need to refactor passing user token as HttpOnly cookie and create a proxy for all the API calls. So I plan to finish the whole feature first after that I can revisit this refactoring if I have time.